testsuites/syncgateway/functional/tests/test_crud.py

```diff
@@ -0,0 +1,275 @@
+from __future__ import print_function
+
+import pytest
+from requests.exceptions import HTTPError
+from couchbase.exceptions import NotFoundError
+from couchbase.bucket import Bucket
+
+from keywords import attachment, document
+from keywords.MobileRestClient import MobileRestClient
+from keywords.SyncGateway import sync_gateway_config_path_for_mode
+from keywords.utils import host_for_url, log_info
+from libraries.testkit.cluster import Cluster
+
+
+@pytest.mark.sanity
+@pytest.mark.syncgateway
+@pytest.mark.xattrs
+@pytest.mark.changes
+@pytest.mark.session
+@pytest.mark.parametrize('sg_conf_name, deletion_type', [
+    ('sync_gateway_default_functional_tests', 'tombstone'),
+    ('sync_gateway_default_functional_tests', 'purge')
+])
+def test_document_resurrection(params_from_base_test_setup, sg_conf_name, deletion_type):
+    """
+    Scenarios:
+
+    Doc meta mode / tombstone
+    - Create docs (set A) via Sync Gateway
+    - Delete docs (set A) via Sync Gateway
+    - Verify docs (set A) are deleted via Sync Gateway
+    - Create docs (set A) via Sync Gateway
+    - Verify revs (set A) are generation 3 via Sync Gateway
+
+    Doc meta mode / purge
+    - Create docs (set A) via Sync Gateway
+    - Purge docs (set A) via Sync Gateway
+    - Verify docs (set A) are deleted via Sync Gateway
+    - Create docs (set A) via Sync Gateway
+    - Verify revs (set A) are generation 1 via Sync Gateway
+
+    XATTRs / tombstone
+    - Create docs (set A) via Sync Gateway
+    - Create docs (set B) via SDK
+    - Delete SDK docs (set B) via Sync Gateway
+    - Delete SG docs (set A) via SDK
+    - Verify docs (set B) are deleted via Sync Gateway
+    - Verify docs (set B) are deleted via SDK
+    - Verify docs (set A) are deleted via Sync Gateway
+    - Verify docs (set A) are deleted via SDK
+    - Create docs (set A) via Sync Gateway
+    - Create docs (set B) via SDK
+    - Verify revs (set A, B) are generation 3 via Sync Gateway
+
+    XATTRs / purge
```

Comment:
Is there a difference between the tombstone and purge scenario from the SDK's perspective?